### PR TITLE
Feature/encrypt quorum comm

### PR DESCRIPTION
**Pull request by julsemaan**
# Description
Encrypt the communications to acquire MariaDB quorum using the clustering password

# Impacts
MariaDB crash recovery

# Delete branch after merge
YES

### CODE REVIEW

File: lib/pf/cluster.pm [hunk 1]
```diff
@@ -644,6 +645,66 @@ sub deactivate_maintenance {
...
+    require pf::config;
+    my $password = $pf::config::Config{active_active}{password};
+    
+    my $desired_length = 56;
+
+    # Ensure it isn't more than 56 bytes
+    $password = substr($password, 0, $desired_length);
+
+    my $missing = $desired_length - length($password) - 1;
+    for my $i (0..$missing) {
```
Comment (extrafu) [inline lib/pf/cluster.pm:664]: eww! how about something cleaner like we could do in C:

foo = calloc(56, sizeof(char));
strncpy(foo, password, strlen(password));

calloc == zero'ed buffer.

Comment (julsemaan) [reply lib/pf/cluster.pm:664]: Doesn't seem like what you said exists in a simple form in perl (from my vast knowledge was acquired across 2 minutes of googling this)

We could do:
```
$password .= '0' x ($desired_length - length($password) - 1)
```

Which would be on a single line and be cleaner

Comment (extrafu) [reply lib/pf/cluster.pm:664]: For advanced Perl questions, ask @lzammit

**Comment by extrafu** [comment]
Reebayise

**Comment by julsemaan** [comment]
ribaysaide

**Pull request merged**